corehq/apps/sso/forms.py

```diff
@@ -583,9 +590,27 @@ class BaseSsoEnterpriseSettingsForm(forms.Form):
         label=gettext_lazy("Entity ID"),
         required=False,
     )
+    enable_user_deactivation = forms.BooleanField(required=False, label="Auto-Deactivation",
+                                help_text="This option ensures any authorization provided by CommCare HQ to"
+                                 " Web Users using SSO is removed when those users are removed from Azure AD"
+                                 " (Entra ID). This includes automatic deactivation of any API keys associated"
+                                 " with these users.",
+                                widget=BootstrapCheckboxInput(inline_label="Automatically deactivate Web Users"))
+    api_host = forms.CharField(required=False, label=gettext_lazy("Tenant Id"))
+    api_id = forms.CharField(required=False, label=gettext_lazy("Application ID"))
+    api_secret = forms.CharField(required=False, label=gettext_lazy("Client Secret"))
+    date_api_secret_expiration = forms.DateTimeField(required=False,
+                                                label=gettext_lazy("Secret Expires On"))
 
     def __init__(self, identity_provider, *args, **kwargs):
+        self.show_remote_user_management = kwargs.pop('show_remote_user_management')
```

Comment:
hey @mjriley `show_user_management` can either be popped out of `kwargs` or used as an argument on the Base form due to the way Django Forms inheritance and init works. It would be less code for @jingcheng16 (and is common practice in our codebase) to follow the `kwargs.pop` route since the OIDC form doesn't require this. For context, this is where the forms are initialized:
https://github.com/dimagi/commcare-hq/blob/a613273eb7480f04873965b634f35de833bf410c/corehq/apps/sso/views/enterprise_admin.py#L109-L121

happy to go to voice on this as explaining through text might be a bit difficult.

Reply:
I think it's possible I'm misunderstanding here. It appears to me that `BaseSsoEnterpriseSettingsForm` requires a `show_remote_user_management` argument, but isn't making that argument explicit. I think we can make that argument explicit, while remaining compatible with the existing code, by using [Keyword-Only Arguments](https://peps.python.org/pep-3102/). The code would look like the following:
```python
    def __init__(self, identity_provider, *args, show_remote_user_management=False, **kwargs):
       self.show_remote_user_management = show_remote_user_management
       ...
       super().__init__(self, *args, **kwargs)
```

Our codebase does have [examples of this syntax](https://github.com/dimagi/commcare-hq/blob/4ecb2aea21d665e8534b929a409f0f6040dcb5f6/corehq/apps/settings/forms.py#L143) (I think Daniel had recommended using this, at the time), and while we also have plenty of forms that use `pop` instead, I'd venture that the primary reason for this is that Keyword-Only Arguments only became available in Python 3.0.

I think it's better to list our arguments explicitly, when possible. Is there a use-case that I'm overlooking here, for why we'd want to keep the `pop` syntax, instead?